Add and install an appdata file (translatable).

badshah400 opened the pull request:
Appdata files are used to generate appstream metadata which is
then used by software stores like gnome-software and plasma-discover
to display the app in its list of available apps for installation.
Basically, installing this appdata file makes your app show up
in these kind of appstream dependent appstores, thus making the
app easily discoverable, installable and reviewable.

The appdata file added here is set up to be translatable using
usual GETTEXT methods, once the appropriate strings are added to
the po files.

See the appdata specification:
https://www.freedesktop.org/software/appstream/docs/

nixxcode commented:
Thanks, good idea!

`make distcheck` [fails](https://travis-ci.org/amsynth/amsynth/jobs/176223342#L1875) with this change though, looks like `amsynth.appdata.xml` is not being cleaned up by make distclean.
```
ERROR: files left in build directory after distclean:
./data/amsynth.appdata.xml
make[1]: *** [distcleancheck] Error 1
```

I can't see any obvious problems in Makefile.am though, the .desktop have a similar rule and don't encounter this problem... any ideas how we can fix that?

badshah400 commented:
Sorry about that! Think the
```
EXTRA_DIST += $(appdata_in_files)
```
line causes problems with distcheck. Let me try to fix this and make a new PR.

badshah400 commented:
Think I fixed that issue and another one, but no idea now why travis is still complaining. At least the make distclean and make check problems are now fixed.

badshah400 commented:
All fixed now, I think!

badshah400 commented:
Happy to be of help :-)

nixxcode commented:
Thanks for the contribution! 😄

badshah400 commented:
Btw, and sorry for keeping on talking in a closed PR, but you may want to add more screenshots to the appdata. I simply used the one over on amsynth's homepage, but larger screenshots showing more features etc. may be useful. Thanks for your awesome app!
